Libraries/Animated/src/animations/SpringAnimation.js

```diff
@@ -192,53 +243,54 @@ class SpringAnimation extends Animation {
-    // The following post provides a lot of thoughts into how to build this
-    // loop: http://gafferongames.com/game-physics/fix-your-timestep/
-    const TIMESTEP_MSEC = 1;
-    const numSteps = Math.floor((now - this._lastTime) / TIMESTEP_MSEC);
-
-    for (let i = 0; i < numSteps; ++i) {
-      // Velocity is based on seconds instead of milliseconds
-      const step = TIMESTEP_MSEC / 1000;
-
-      // This is using RK4. A good blog post to understand how it works:
-      // http://gafferongames.com/game-physics/integration-basics/
-      const aVelocity = velocity;
-      const aAcceleration =
-        this._tension * (this._toValue - tempPosition) -
-        this._friction * tempVelocity;
-      tempPosition = position + aVelocity * step / 2;
-      tempVelocity = velocity + aAcceleration * step / 2;
-
-      const bVelocity = tempVelocity;
-      const bAcceleration =
-        this._tension * (this._toValue - tempPosition) -
-        this._friction * tempVelocity;
-      tempPosition = position + bVelocity * step / 2;
-      tempVelocity = velocity + bAcceleration * step / 2;
-
-      const cVelocity = tempVelocity;
-      const cAcceleration =
-        this._tension * (this._toValue - tempPosition) -
-        this._friction * tempVelocity;
-      tempPosition = position + cVelocity * step / 2;
-      tempVelocity = velocity + cAcceleration * step / 2;
-
-      const dVelocity = tempVelocity;
-      const dAcceleration =
-        this._tension * (this._toValue - tempPosition) -
-        this._friction * tempVelocity;
-      tempPosition = position + cVelocity * step / 2;
-      tempVelocity = velocity + cAcceleration * step / 2;
-
-      const dxdt = (aVelocity + 2 * (bVelocity + cVelocity) + dVelocity) / 6;
-      const dvdt =
-        (aAcceleration + 2 * (bAcceleration + cAcceleration) + dAcceleration) /
-        6;
-
-      position += dxdt * step;
-      velocity += dvdt * step;
+    let deltaTime = 0.0;
+    if (now > this._lastTime) {
+      deltaTime = (now - this._lastTime) / 1000;
+    }
+    this._frameTime += deltaTime;
+
+    const c: number = this._damping || 0;
+    const m: number = this._mass || 0;
+    const k: number = this._stiffness || 0;
+    const v0: number = -(this._initialVelocity || 0);
+
+    invariant(m > 0, 'Mass value must be greater than 0');
+    invariant(k > 0, 'Stiffness value must be greater than 0');
+    invariant(c > 0, 'Damping value must be greater than 0');
+
+    const zeta = c / (2 * Math.sqrt(k * m)); // damping ratio
+    const omega0 = Math.sqrt(k / m); // undamped angular frequency of the oscillator (rad/ms)
+    const omega1 = omega0 * Math.sqrt(1.0 - zeta * zeta); // exponential decay
+    const x0 = this._toValue - this._startPosition; // calculate the oscillation from x0 = 1 to x = 0
+
+    let position = 0.0;
+    let velocity = 0.0;
+    const t = this._frameTime;
+    if (zeta < 1) {
+      // Under damped
+      const envelope = Math.exp(-zeta * omega0 * t);
+      position =
```

Comment:
I think we should `// prettier-ignore` those and format manually it looks kind of weird, especially the velocity line haha